### PR TITLE
add new sensor for ip6 addresses

### PR DESCRIPTION
**Pull request by h3mmy**
## Summary

This PR is intended to address #4198: Adds a new network sensor to detect ipv6 addresses and list them as attributes

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#1031

## Any other notes

I'm still trying to figure out how best to test this. As such I'm leaving this as a draft for now

### CODE REVIEW

File: common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt [hunk 1]
```diff
@@ -148,7 +161,7 @@ class NetworkSensorManager : SensorManager {
             wifiSignalStrength
         )
         val list = if (hasWifi(context)) {
-            val withPublicIp = wifiSensors.plus(publicIp)
+            val withPublicIp = wifiSensors.plus(publicIp).plus(ip6Addresses)
```
Comment (dshokouhi) [inline common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt:164]: this sensor requires the user to have Android 6 or higher so we need to make sure we do not make the sensor available to all user as the app supports Android 5+

File: common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt [hunk 2]
```diff
@@ -356,6 +369,40 @@ class NetworkSensorManager : SensorManager {
...
+        }
+        val ipAddressList: List<String> = ArrayList()
+        var totalAddresses = 0
+
+        if (checkPermission(context, ip6Addresses.id)) {
+            val connectivityManager = context.applicationContext.getSystemService<ConnectivityManager>()
+            val activeNetwork = connectivityManager?.activeNetwork
+            val ipAddresses = connectivityManager?.getLinkProperties(activeNetwork)?.linkAddresses
+            if (ipAddresses.isNullOrEmpty()) {
+                return
```
Comment (dshokouhi) [inline common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt:385]: `return` does not seem correct here as we need to make sure to actually update the sensor to send an appropriate state if the sensor is enabled with proper permissions

File: common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt [hunk 3]
```diff
@@ -119,6 +120,18 @@ class NetworkSensorManager : SensorManager {
...
         )
+        val ip6Addresses = SensorManager.BasicSensor(
+            "ip6_addresses",
+            "sensor",
+            commonR.string.basic_sensor_name_ip6_addresses,
+            commonR.string.sensor_description_ip6_addresses,
+            "mdi:ip",
+            unitOfMeasurement = "connection(s)",
+            stateClass = SensorManager.STATE_CLASS_MEASUREMENT,
+            docsLink = "https://companion.home-assistant.io/docs/core/sensors#connection-type-sensor",
```
[the remark this answers is not in the document]
Comment (h3mmy) [reply common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt]: Thanks for pointing that out. Fixed

File: common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt (hunk 1 above)
Comment (h3mmy) [reply common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt]: Yes, I fixed that in a later commit. The change should be in here now

File: common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt (hunk 2 above)
Comment (h3mmy) [reply common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt]: copy-paste error. The only return now is if it is disabled

**Comment by h3mmy** [comment]
I'm marking this as ready for review, but I'd like to do some more manual testing before it is considered merge-able.

**Comment by h3mmy** [comment]
The PR is ready. I've verified that existing functionality is not impacted with some cursory testing. 

[x] Sensor does not update when disabled
[x] Sensor displays `0 address(es)` when no ipv6
[x] Sensor displays correct count of ipv6 addresses bound to active interface
[x] Sensor contains list of addresses in attribute
[x] IPv4 addresses are excluded from the list and also the count

Screenshot_20240213-200826
Screenshot_20240213-200819

**Review by dshokouhi — APPROVED** [review]
Looks good to me, tested the new sensor and its working as expected

**Review by JBassett — APPROVED** [review]

**Pull request merged**